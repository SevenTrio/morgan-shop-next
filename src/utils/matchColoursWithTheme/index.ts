import { matchColoursWithTheme } from './matchColoursWithTheme';

export default matchColoursWithTheme;
