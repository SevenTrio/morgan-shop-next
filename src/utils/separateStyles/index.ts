import { separateStyles } from './separateStyles';

export default separateStyles;
