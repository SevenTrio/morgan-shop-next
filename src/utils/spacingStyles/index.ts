import { spacingStyles } from './spacingStyles';

export default spacingStyles;
