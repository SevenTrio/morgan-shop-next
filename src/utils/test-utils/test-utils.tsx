import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';

import { GlobalStyle } from 'styles/global-styles';
import { theme } from 'styles/theme';

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ThemeProvider theme={theme}>
    <Normalize />
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
