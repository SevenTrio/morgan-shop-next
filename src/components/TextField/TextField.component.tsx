import { forwardRef, InputHTMLAttributes, ReactNode, useState } from 'react';

import { ISpacingProps } from 'types/spacing.types';

import separateStyles from 'utils/separateStyles';

import Typography from 'components/Typography';

import {
  StyledTextField,
  StyledHints,
  StyledInputWrapper,
  StyledInput,
  ShowPasswordButton,
  StyledEyeOnIcon,
  StyledEyeOffIcon,
} from './TextField.styles';

export interface ITextFieldProps
  extends ISpacingProps,
    InputHTMLAttributes<HTMLInputElement> {
  /**
   * The id of the input element.
   * Use this prop to make label and helperText accessible for screen readers.
   */
  id?: string;
  /**
   * The label content.
   */
  label?: ReactNode;
  /**
   * The variant to use.
   */
  action?: ReactNode;
  /**
   * The helper text content.
   */
  helperText?: ReactNode;
  /**
   * If true, the label will be displayed in an error state.
   */
  error?: boolean;
}

export const TextField = forwardRef<HTMLInputElement, ITextFieldProps>(
  (
    {
      id,
      label,
      action,
      helperText,
      error = false,
      type = 'text',
      ...restProps
    },
    ref,
  ) => {
    const [showPassword, setShowPassword] = useState(false);
    const isPasswordField = type === 'password';

    const [wrapperStyles, inputProps] = separateStyles(restProps);

    const handleShowPassword = () => {
      setShowPassword(true);
    };

    const handleHidePassword = () => {
      setShowPassword(false);
    };

    return (
      <StyledTextField
        withHelperText={!!helperText}
        isPasswordField={isPasswordField}
        {...wrapperStyles}
      >
        <StyledHints>
          {label ? (
            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
            // @ts-ignore
            <Typography component="label" variant="body2" htmlFor={id}>
              {label}
            </Typography>
          ) : null}

          {action}
        </StyledHints>

        <StyledInputWrapper>
          <StyledInput
            id={id}
            ref={ref}
            type={isPasswordField && !showPassword ? 'password' : 'text'}
            error={error}
            isPasswordField={isPasswordField}
            {...inputProps}
          />

          {isPasswordField && (
            <ShowPasswordButton>
              {showPassword ? (
                <StyledEyeOnIcon onClick={handleHidePassword} />
              ) : (
                <StyledEyeOffIcon onClick={handleShowPassword} />
              )}
            </ShowPasswordButton>
          )}
        </StyledInputWrapper>

        {helperText ? (
          <Typography
            variant="caption1"
            color={error ? 'error' : 'textSecondary'}
          >
            {helperText}
          </Typography>
        ) : null}
      </StyledTextField>
    );
  },
);
