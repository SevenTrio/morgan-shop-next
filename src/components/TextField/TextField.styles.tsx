import styled, { css } from 'styled-components';

import { ISpacingProps } from 'types/spacing.types';

import spacingStyles from 'utils/spacingStyles';

import EyeOnIcon from 'icons/ic_eye_on.svg';
import EyeOffIcon from 'icons/ic_eye_off.svg';

interface IStyledTextFieldProps extends ISpacingProps {
  withHelperText: boolean;
  isPasswordField: boolean;
}

interface IStyledInputProps {
  error: boolean;
  isPasswordField: boolean;
}

export const StyledTextField = styled.div<IStyledTextFieldProps>`
  ${spacingStyles};

  display: flex;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.sm}px) {
    margin-bottom: 24px;
  }

  ${({ withHelperText }) =>
    withHelperText &&
    css`
      margin-bottom: 12px;

      @media screen and (max-width: ${({ theme }) => theme.breakpoints.sm}px) {
        margin-bottom: 8px;
      }
    `}
`;

export const StyledHints = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const StyledInputWrapper = styled.div`
  position: relative;
`;

export const StyledInput = styled.input<IStyledInputProps>`
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid ${({ theme }) => theme.palette.common.darkGray};
  font-family: 'Poppins', sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: ${({ theme }) => theme.palette.text.primary};
  background-color: ${({ theme }) => theme.palette.background.default};

  &:focus {
    border: 1px solid ${({ theme }) => theme.palette.primary.main};
  }

  &::placeholder {
    font-weight: 400;
    color: ${({ theme }) => theme.palette.text.hint};
  }

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.sm}px) {
    padding: 0 8px;
    font-size: 0.75rem;
  }

  ${({ isPasswordField }) =>
    isPasswordField &&
    css`
      padding-right: 48px;
    `}

  ${({ error }) =>
    error &&
    css`
      border: 1px solid ${({ theme }) => theme.palette.error.main};
    `}
`;

export const ShowPasswordButton = styled.button`
  position: absolute;
  right: 8px;
  top: 50%;
  transform: translate(0, -50%);
  display: flex;
  flex-shrink: 0;
  justify-content: center;
  align-items: center;
  width: 32px;
  height: 32px;
  padding: 0;
  border: none;
  border-radius: 50%;
  background: none;
  cursor: pointer;
`;

export const StyledEyeOnIcon = styled(EyeOnIcon)`
  width: 24px;
  height: 24px;

  path {
    stroke: ${({ theme }) => theme.palette.text.secondary};
  }
`;

export const StyledEyeOffIcon = styled(EyeOffIcon)`
  width: 24px;
  height: 24px;

  path {
    stroke: ${({ theme }) => theme.palette.text.secondary};
  }
`;
