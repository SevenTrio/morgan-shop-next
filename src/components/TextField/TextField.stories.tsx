import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { TextField } from './TextField.component';

export default {
  title: `Components/TextField`,
  component: TextField,
} as ComponentMeta<typeof TextField>;

const Template: ComponentStory<typeof TextField> = (args) => (
  <TextField {...args} />
);

export const Default = Template.bind({});
Default.args = {
  label: `Default`,
  placeholder: `Default`,
};

export const Password = Template.bind({});
Password.args = {
  label: `Password`,
  placeholder: `Password`,
  type: `password`,
};

export const HelperText = Template.bind({});
HelperText.args = {
  label: `Helper text`,
  placeholder: `Helper text`,
  helperText: `Some important text`,
};

export const Error = Template.bind({});
Error.args = {
  label: `Error`,
  placeholder: `Error`,
  helperText: `Incorrect entry.`,
  error: true,
};
