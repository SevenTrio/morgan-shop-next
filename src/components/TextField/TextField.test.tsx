import { ChangeEvent, FC, useState } from 'react';
import { render, fireEvent } from 'utils/test-utils';

import { ITextFieldProps } from 'components/TextField/TextField.component';

import TextField from 'components/TextField';

const ControlledTextField: FC = (props?: Partial<ITextFieldProps>) => {
  const [value, setValue] = useState('');

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };

  return <TextField value={value} onChange={handleChange} {...props} />;
};

const setupControlledTextField = (props?: Partial<ITextFieldProps>) => {
  const utils = render(<ControlledTextField aria-label="input" {...props} />);

  const input = utils.getByLabelText('input') as HTMLInputElement;

  return {
    input,
    ...utils,
  };
};

const setupUncontrolledTextField = (props?: Partial<ITextFieldProps>) => {
  const utils = render(<TextField aria-label="input" {...props} />);

  const input = utils.getByLabelText('input') as HTMLInputElement;

  return {
    input,
    ...utils,
  };
};

describe(`TextField component`, () => {
  describe(`renders correctly`, () => {
    test(`without props`, () => {
      const { container } = render(<TextField />);
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
    });

    test(`with label prop`, () => {
      const labelText = `Lorem ipsum dolor.`;
      const { container, getByText } = render(<TextField label={labelText} />);
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
      expect(getByText(labelText)).not.toBeNull();
    });

    test(`with helperText prop`, () => {
      const helperText = `Lorem ipsum dolor.`;
      const { container, getByText } = render(
        <TextField helperText={helperText} />,
      );
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
      expect(getByText(helperText)).not.toBeNull();
    });

    test(`with action prop`, () => {
      const actionText = `Lorem ipsum dolor.`;
      const { container, getByText } = render(
        <TextField action={<p>{actionText}</p>} />,
      );
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
      expect(getByText(actionText)).not.toBeNull();
    });

    test(`with type prop`, () => {
      const { container, input } = setupUncontrolledTextField({
        type: 'password',
      });
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
      expect(input.type).toBe('password');
    });
  });

  describe(`applies correct styles`, () => {
    test(`to wrapper component`, () => {
      const { container } = render(
        <TextField m={2} style={{ display: 'none' }} className="Test" />,
      );
      const tree = container.firstChild as HTMLElement;

      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule(`margin`, `8px`);
      expect(tree?.style.display).toBe('none');
      expect(tree?.className.split(' ').includes('Test')).toBe(true);
    });
  });

  describe(`retains the value correctly`, () => {
    test(`when controlled`, () => {
      const inputValue = `Lorem ipsum dolor.`;
      const { container, input } = setupControlledTextField();
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
      expect(input.value).toBe('');
      fireEvent.change(input, { target: { value: inputValue } });
      expect(input.value).toBe(inputValue);
    });

    test(`when uncontrolled`, () => {
      const inputValue = `Lorem ipsum dolor.`;
      const { container, input } = setupUncontrolledTextField();
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
      expect(input.value).toBe('');
      fireEvent.change(input, { target: { value: inputValue } });
      expect(input.value).toBe(inputValue);
    });
  });
});
