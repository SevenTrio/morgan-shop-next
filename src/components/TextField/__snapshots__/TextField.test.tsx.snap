// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component applies correct styles to wrapper component 1`] = `
.c0 {
  margin: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid #6D6D6D;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: #121212;
  background-color: #FAFAFA;
}

.c3:focus {
  border: 1px solid #121212;
}

.c3::-webkit-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::-moz-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3:-ms-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (max-width:600px) {
  .c3 {
    padding: 0 8px;
    font-size: 0.75rem;
  }
}

<div
  class="c0 Test"
  style="display: none;"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <input
      class="c3"
      type="text"
    />
  </div>
</div>
`;

exports[`TextField component renders correctly with action prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid #6D6D6D;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: #121212;
  background-color: #FAFAFA;
}

.c3:focus {
  border: 1px solid #121212;
}

.c3::-webkit-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::-moz-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3:-ms-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (max-width:600px) {
  .c3 {
    padding: 0 8px;
    font-size: 0.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <p>
      Lorem ipsum dolor.
    </p>
  </div>
  <div
    class="c2"
  >
    <input
      class="c3"
      type="text"
    />
  </div>
</div>
`;

exports[`TextField component renders correctly with helperText prop 1`] = `
.c4 {
  padding: 0;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #6D6D6D;
  text-align: inherit;
  font-size: 0.75rem;
  line-height: 1.6666666666;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;
  margin-bottom: 12px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid #6D6D6D;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: #121212;
  background-color: #FAFAFA;
}

.c3:focus {
  border: 1px solid #121212;
}

.c3::-webkit-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::-moz-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3:-ms-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

@media screen and (max-width:600px) {
  .c4 {
    font-size: 0.625rem;
    line-height: 1.6;
  }
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 8px;
  }
}

@media screen and (max-width:600px) {
  .c3 {
    padding: 0 8px;
    font-size: 0.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <input
      class="c3"
      type="text"
    />
  </div>
  <span
    class="c4"
    color="textSecondary"
  >
    Lorem ipsum dolor.
  </span>
</div>
`;

exports[`TextField component renders correctly with label prop 1`] = `
.c2 {
  padding: 0;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #121212;
  text-align: inherit;
  font-size: 0.875rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid #6D6D6D;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: #121212;
  background-color: #FAFAFA;
}

.c4:focus {
  border: 1px solid #121212;
}

.c4::-webkit-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c4::-moz-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c4:-ms-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c4::placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (max-width:600px) {
  .c4 {
    padding: 0 8px;
    font-size: 0.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <label
      class="c2"
      color="textPrimary"
    >
      Lorem ipsum dolor.
    </label>
  </div>
  <div
    class="c3"
  >
    <input
      class="c4"
      type="text"
    />
  </div>
</div>
`;

exports[`TextField component renders correctly with type prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid #6D6D6D;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: #121212;
  background-color: #FAFAFA;
  padding-right: 48px;
}

.c3:focus {
  border: 1px solid #121212;
}

.c3::-webkit-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::-moz-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3:-ms-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c4 {
  position: absolute;
  right: 8px;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 32px;
  height: 32px;
  padding: 0;
  border: none;
  border-radius: 50%;
  background: none;
  cursor: pointer;
}

.c5 {
  width: 24px;
  height: 24px;
}

.c5 path {
  stroke: #6D6D6D;
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (max-width:600px) {
  .c3 {
    padding: 0 8px;
    font-size: 0.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <input
      aria-label="input"
      class="c3"
      type="password"
    />
    <button
      class="c4"
    >
      <svg
        class="c5"
      />
    </button>
  </div>
</div>
`;

exports[`TextField component renders correctly without props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid #6D6D6D;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: #121212;
  background-color: #FAFAFA;
}

.c3:focus {
  border: 1px solid #121212;
}

.c3::-webkit-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::-moz-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3:-ms-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (max-width:600px) {
  .c3 {
    padding: 0 8px;
    font-size: 0.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <input
      class="c3"
      type="text"
    />
  </div>
</div>
`;

exports[`TextField component retains the value correctly when controlled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid #6D6D6D;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: #121212;
  background-color: #FAFAFA;
}

.c3:focus {
  border: 1px solid #121212;
}

.c3::-webkit-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::-moz-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3:-ms-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (max-width:600px) {
  .c3 {
    padding: 0 8px;
    font-size: 0.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <input
      aria-label="input"
      class="c3"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`TextField component retains the value correctly when uncontrolled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  padding: 0 12px;
  border: 1px solid #6D6D6D;
  font-family: 'Poppins',sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  color: #121212;
  background-color: #FAFAFA;
}

.c3:focus {
  border: 1px solid #121212;
}

.c3::-webkit-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::-moz-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3:-ms-input-placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

.c3::placeholder {
  font-weight: 400;
  color: #6D6D6D;
}

@media screen and (max-width:600px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (max-width:600px) {
  .c3 {
    padding: 0 8px;
    font-size: 0.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <input
      aria-label="input"
      class="c3"
      type="text"
    />
  </div>
</div>
`;
